scalafmt-core/shared/src/main/scala/org/scalafmt/util/GitOps.scala

```diff
@@ -16,42 +16,39 @@ object GitOps {
   def apply(): GitOps = new GitOpsImpl(AbsoluteFile.userDir)
 }
 
-class GitOpsImpl(workingDirectory: AbsoluteFile) extends GitOps {
+class GitOpsImpl(private[util] val workingDirectory: AbsoluteFile)
+    extends GitOps {
 
-  def exec(cmd: Seq[String]): String = {
-    val lastError = new StringBuilder
-    val swallowStderr = ProcessLogger(_ => Unit, err => lastError.append(err))
-    try {
-      sys.process
-        .Process(cmd, workingDirectory.jfile)
-        .!!(swallowStderr)
-        .trim
-    } catch {
-      case NonFatal(e) =>
-        throw new IllegalStateException(
-          s"Failed to run command ${cmd.mkString(" ")}. " +
-            s"Error: ${lastError.toString()}",
-          e)
+  private[util] def exec(cmd: Seq[String]): Try[Seq[String]] = {
+    val gitRes: Try[String] = Try {
+      val lastError = new StringBuilder
+      val swallowStderr = ProcessLogger(_ => Unit, err => lastError.append(err))
+      try {
+        sys.process
```

Comment:
Wait, can we use Java's process builder? Won't that interfere with the ScalaNative push?